### DISCUSSION TITLE
Allow deployment scripts

### DISCUSSION THREAD

**Discussion by Saifallak** [coollabsio/coolify#3611] [Improvement Requests]
### Description

we might need to excute custom commands after/before/during build
can we have something like deployment scripts ?

in easypanel.io we have it.

![image](https://github.com/user-attachments/assets/7df70b9f-319d-480d-b4aa-0f6c1967ecb7)

### Minimal Reproduction (if possible, example repository)

none

### Exception or Error

none

### Version

v4.0.0-beta.315

**Comment by DarkGhostHunter** [comment]
You mean this?

![image](https://github.com/user-attachments/assets/6b856343-6757-4a19-abf3-005fcf8a6231)

**Comment by Saifallak** [comment]
> You mean this?
> 
> ![image](https://private-user-images.githubusercontent.com/5141911/349783069-6b856343-6757-4a19-abf3-005fcf8a6231.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyOTMzMzksIm5iZiI6MTcyMTI5MzAzOSwicGF0aCI6Ii81MTQxOTExLzM0OTc4MzA2OS02Yjg1NjM0My02NzU3LTRhMTktYWJmMy0wMDVmY2Y4YTYyMzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDg1NzE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGNjZWEwZDAxY2RkMjM1YzNhMDYzYzljZDkwZTZmZjMxMzdjYThkYzE3MjFiMmQ2ZDAwNDU5ZjRhNTFhOTkzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.whj3G6p3NQA-Iq85lvgmO6DFKMVso55i7WP1Ak1k1A4)

Not quite right.
Pre deployment runs on old container, to turn off stuff i guess ..
Post deployment, runs on **new** container after every thing finished (eg. Build) ..

I want something before build in the new container
And after build ..

Or maybe easier to customize whole deployment script to edit whatever we like

**Comment by DarkGhostHunter** [comment]
I think you should edit the Dockerfile

**Comment by Saifallak** [comment]
> I think you should edit the Dockerfile

Where do i find it? If its just simple gh repo clone

**Comment by DarkGhostHunter** [comment]
> > I think you should edit the Dockerfile
> 
> Where do i find it? If its just simple gh repo clone

What's your deployment? Nixpacks? Dockerfile? Docker compose? Static?
